src/eth1spec/base_types.py

```
@@ -33,6 +33,9 @@ def from_be_bytes(cls: Type, buffer: "Bytes") -> "Uint":
         self : `Uint`
             Unsigned integer decoded from `buffer`.
         """
+        if len(buffer) > 1 and buffer[0] == 0:
+            raise ValueError("Big endian representation can't start with 0")
+
```

I'm curious where this restriction comes from?

---

This actually is coming from the Ethereum RLP fixtures. It is expected that RLP decoding should throw an error, if the data to decode (or any intermediate data which is to be length derived from big endian representation) is not a big endian representation.

---

I'm not sure I understand why this would be an error in `Uint`. Normally for `int` I can say `int(0x0001)`. That should also still be allowed for `Uint`. If it's an RLP requirement, I think it should checked within the RLP module.

---

I have reverted this specific change and checked that the be bytes don't start with 0 in the rlp module